addons/loyalty/models/product_pricelist.py

```diff
@@ -0,0 +1,12 @@
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    def action_archive(self):
+        loyalty_programs = self.env['loyalty.program'].search([('pricelist_ids', 'in', self.ids)], limit=1)
```

Comment:
```suggestion
        loyalty_programs = self.env['loyalty.program'].search([
           ('pricelist_ids', 'in', self.ids),
       ], limit=1)
```
We try to keep lines to 100 chars max in this scope